Sources/RoutingKit/TrieRouter/TrieRouterBuilder.swift

```diff
@@ -0,0 +1,111 @@
+import Logging
+
+public struct TrieRouterBuilder<Output: Sendable>: RouterBuilder {
+    typealias Node = TrieRouterNode<Output>
+
+    /// Configured options such as case-sensitivity.
+    public let options: Set<TrieRouter<Output>.ConfigurationOption>
+
+    /// Configured logger.
+    public let logger: Logger
+
+    var root: Node
```

Comment:
`Node` is a class. Why, then, does this need to be a `var` instead of a `let`?

Reply:
Because when we register a new node we essentially rebuild the whole tree and store it in `root` as a completely new thing. Even though it's a class we're not mutating the node itself